caddyhttp/httpserver/replacer.go

```diff
@@ -119,6 +122,19 @@ func NewReplacer(r *http.Request, rr *ResponseRecorder, emptyValue string) Repla
 
 				return requestReplacer.Replace(string(dump))
 			},
+			"{request_body}": func() string {
+				if !canLogRequest(r) {
+					return ""
+				}
+
+				body, err := readRequestBody(r)
+				if err != nil {
+					log.Printf("[WARNING] Cannot copy request body %v", err)
+					return ""
+				}
+
+				return string(body)
```

Comment:
Might be good to condense newlines into \n and \r literals. See `requestReplacer` in this file.